src/components/Restoration/SummaryRestoreBusiness.vue

```diff
@@ -0,0 +1,179 @@
+            <label>Restoration Type</label>
+          </v-col>
+
+          <v-col cols="12" sm="9">
+            <template v-if="isLimitedRestoration">
+              <p id="limited-restoration">Limited Restoration</p>
+              <p id="expires-on">Expires on {{ expiry }}</p>
+            </template>
+            <template v-else>
+              <p id="full-restoration">Full Restoration</p>
+              <p id="applicants-relationships">Applicant's relationship(s): {{ relationships }}.</p>
+            </template>
+          </v-col>
+        </v-row>
+
+        <!-- Approval Type -->
+        <v-row no-gutters class="mt-6">
+          <v-col cols="12" sm="3" class="pr-4">
+            <label>Approval Type</label>
+          </v-col>
+
+          <v-col cols="12" sm="9">
+            <template v-if="getRestoration.approvalType === ApprovalTypes.VIA_COURT_ORDER">
+              <label>Approved by Court Order</label><br/>
+              <p id="court-order-number">Court Order Number: {{ getRestoration.courtOrder.fileNumber }}</p>
+            </template>
+
+            <template v-else-if="getRestoration.approvalType === ApprovalTypes.VIA_REGISTRAR">
+              <label>Approved by Registrar</label>
+              <p id="notice-date">BC Gazette publish date: {{ noticeDate }}</p>
+              <p id="application-date">Application for Restoration mailed date: {{ applicationDate }}</p>
+            </template>
+          </v-col>
+        </v-row>
+      </article>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Watch } from 'vue-property-decorator'
+import { DateMixin } from '@/mixins'
+import { Getter, Action } from 'vuex-class'
+import { ApprovalTypes, CorpTypeCd, RouteNames, RestorationTypes } from '@/enums'
+import { NameTranslationIF, RestorationStateIF } from '@/interfaces'
+import { GetCorpFullDescription } from '@bcrs-shared-components/corp-type-module'
+
+@Component({
+  mixins: [
+    DateMixin
+  ]
+})
+export default class SummaryRestoreBusiness extends Vue {
+  @Getter getEntityType!: CorpTypeCd
+  @Getter getNameRequestApprovedName!: string
+  @Getter getNameTranslations!: NameTranslationIF[]
+  @Getter getRestoration!: RestorationStateIF
+  @Getter isRestoreBusinessNameValid!: boolean
+
+  // for template
+  readonly RouteNames = RouteNames
+  readonly ApprovalTypes = ApprovalTypes
+
+  /** The entity description. */
+  get entityDescription (): string {
+    return GetCorpFullDescription(this.getEntityType)
+  }
+
+  get isLimitedRestoration (): boolean {
+    return (this.getRestoration.type === RestorationTypes.LIMITED)
+  }
+
+  get expiry (): string {
+    return this.yyyyMmDdToPacificDate(this.getRestoration.expiry)
+  }
+
+  get relationships (): string {
+    return this.getRestoration.relationships.join(', ')
+  }
+
+  get noticeDate (): string {
+    return this.yyyyMmDdToPacificDate(this.getRestoration.noticeDate)
+  }
+
+  get applicationDate (): string {
```

Comment:
Much cleaner than what I've previously done. Having all these computed value getters instead of inserting everything directly into the template. I'll be adapting this approach moving forward.

Reply:
Jonathan also commented elsewhere that these are now testable in code (not looking at element text).

👍

Reply:
@JazzarKarim OK to resolve this conversation?